### ISSUE TITLE
support alternative output formats for DescribeFeatureType WFS request

### ISSUE THREAD

**Issue by landryb** [MapServer/MapServer#6190]
some clients assume that if `application/json` is supported in `GetFeature` then it will be supported by other operations, which alas isnt the case, per https://github.com/MapServer/MapServer/blob/master/mapwfs.cpp#L81 and https://github.com/MapServer/MapServer/blob/master/mapwfs.cpp#L1311.

akin to `wfs_getfeature_formatlist` would be nice to have `wfs_describefeaturetype_formatlist` or at least get `application/json` supported by default, as it seems to be used more and more in web viewers (which tend to thing everything is geoserver...)

**Comment by jratike80** [comment]
Which clients assume that? I would not encourage any client developers to trust on that but stay with GML if they want to build interoperable systems. Geoservers, as far as I know, do not support JSON for anything else than as an outputformat http://demo.geo-solutions.it/geoserver/wfs?service=WFS&version=2.0.0&request=getcapabilities. Documentation is at https://docs.geoserver.org/latest/en/user/services/wfs/outputformats.html 

> Note outputFormat=json is only supported for getFeature (for backward compatibility).

JSON based OGC API services are gaining popularity and Mapserver has a support for OAPI Features in a development row.

**Comment by landryb** [comment]
geo-solutions/mapstore2 is heavily using `application/json` in `DescribeFeatureType` WFS query to get the structure of a remote layer - without checking if it's in the list of supported formats for DescribeFeatureType (but that's also the case for GetFeatureInfo and GetMap, per georchestra/mapstore2-georchestra#301 and georchestra/mapstore2-georchestra#300).

I agree that geoserver doesnt list it in the supported output formats for `DescribeFeatureType` operation, but it replies just fine, so maybe that a hidden feature but it's used..

https://ids.craig.fr/wxs/geor_loc/wfs?service=WFS&version=1.1.0&request=describefeaturetype&typename=geor_loc:communes_201806&outputformat=application/json for example is a geoserver 2.16 with a default setup, the one at https://demo.geo-solutions.it/geoserver/ is an old 2.12 with broken layers.

great that OAPIF support is coming as a first-class citizen :)

**Comment by jratike80** [comment]
Right, application/json is accepted by Geoserver 2.18. I need to fix the documentation and check if there is already a ticket about fixing the GetCapabilities to reflect the reality.
Test query as a note to myself:
http://localhost:8080/geoserver/wfs?service=WFS&version=2.0.0&request=DescribeFeatureType&typenames=topp:states&outputformat=application/json
